presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -36,40 +36,37 @@
 
 public class RuleAssert
 {
-    private final Metadata metadata;
-    private Session session;
     private final Rule rule;
 
     private final PlanNodeIdAllocator idAllocator = new PlanNodeIdAllocator();
+    private final LocalQueryRunner queryRunner;
```

Comment:
Why do we need the full LocalQueryRunner? I've been trying to keep the rule assertion framework as agnostic to it as possible. Maybe we can just take the "in transaction" logic from it and copy it here? I don't think it does much anyway:

```java
    public <T> T inTransaction(Function<Session, T> transactionSessionConsumer)
    {
        return inTransaction(defaultSession, transactionSessionConsumer);
    }

    public <T> T inTransaction(Session session, Function<Session, T> transactionSessionConsumer)
    {
        return transaction(transactionManager, accessControl)
                .singleStatement()
                .execute(session, transactionSessionConsumer);
    }
```